nix_update/version/github.py

```diff
@@ -19,11 +30,10 @@ def fetch_github_version(url: ParseResult) -> Optional[str]:
     info(f"fetch {feed_url}")
     resp = urllib.request.urlopen(feed_url)
     tree = ET.fromstring(resp.read())
-    release = tree.find(".//{http://www.w3.org/2005/Atom}entry")
-    if release is None:
-        raise VersionError("No release found")
-    link = release.find("{http://www.w3.org/2005/Atom}link")
-    assert link is not None
-    href = link.attrib["href"]
-    url = urlparse(href)
-    return url.path.split("/")[-1]
+    releases = tree.findall(".//{http://www.w3.org/2005/Atom}entry")
+    entries = [version_from_entry(x) for x in releases]
+    filtered = [
+        x for x in entries if x is not None and extract_version(x, version_extractor)
```

Comment:
It should be actually possible to interfere prefixes and suffixes by looking at difference between `version` and rev in `src` but you have a point there in also using this regex to find github releases.